frontend/src/components/core/Sidebar/SidebarNav.tsx

```diff
@@ -57,7 +57,18 @@ const SidebarNav = ({
     return null
   }
 
-  const [expanded, setExpanded] = useState(false)
+  // Check localStorage to see if the user has a preference set
+  const isLocalStorageSet = localStorage.getItem("navExpanded") !== null
+  const shouldNavExpand =
+    isLocalStorageSet && localStorage.getItem("navExpanded") === "true"
+  const pageLimit = shouldNavExpand === true ? -1 : 6
+
+  const [expanded, setExpanded] = useState(
+    hasSidebarElements === true ? false : shouldNavExpand
+  )
+  const [pagesToShow, setPagesToShow] = useState(
+    hasSidebarElements === true ? 6 : pageLimit
+  )
```

Comment:
I think it could be helpful to add some explanatory comments to the default values here (ex: If there are sidebar elements, we want to render a collapsed MPA page nav. If not, we defer to the localStorage preference)